src/controllers/auth.controller.js

```diff
@@ -0,0 +1,141 @@
+import { User } from "../models/user.js";
+import { userService } from "../services/user.service.js";
+import { jwtService } from "../services/jwt.service.js";
+import { ApiError } from "../exeptions/api.error.js";
+import bcrypt from 'bcrypt';
+import { toeknService } from "../services/token.service.js";
+
+function validateEmail(value) {
+  if (!value) {
+    return 'Email is required';
+  }
+
+  const emailPattern = /^[\w.+-]+@([\w-]+\.){1,3}[\w-]{2,}$/;
+
+  if (!emailPattern.test(value)) {
+    return 'Email is not valid';
+  }
+}
+
+function validatePassword(value) {
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+}
+
+async function generateTokens(res, user) {
+  const normalizedUser = userService.normalize(user);
+  const accessToken = jwtService.sign(normalizedUser);
+  const refreshToken = jwtService.signRefresh(normalizedUser);
+
+  await toeknService.save(normalizedUser.id, refreshToken);
+
+  res.cookie('refreshToken', refreshToken, {
+    maxAge: 30 * 24 * 60 * 60 * 1000,
+    HttpOnly: true,
+  });
+  res.send({
+    user: normalizedUser,
+    accessToken
+  });
+}
+
+const register = async (req, res) => {
+  const { email, password } = req.body;
+
+  const errors = {
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (errors.email || errors.password) {
+    throw ApiError.badRequest('Bad request', errors);
```

Comment:
Exposing detailed error messages like 'Bad request' with specific field errors can be a security risk. Consider using more generic error messages to avoid revealing sensitive information.